### PR TITLE
feat: prometheus metrics

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: https://github.com/lightdash/lightdash-cloud/issues/401

### Description:

Add Prometheus metrics endpoint. 

### Prometheus environment variables

| Variable                                    | Description                                                                     | Required? | Default                     |
| ------------------------------------------- | ------------------------------------------------------------------------------- | --------- | --------------------------- |
| `LIGHTDASH_PROMETHEUS_ENABLED`              | Enables/Disables Prometheus metrics endpoint                                    |           | `false`                     |
| `LIGHTDASH_PROMETHEUS_PORT`                 | Port for Prometheus metrics endpoint                                            |           | `9090`                      |
| `LIGHTDASH_PROMETHEUS_PATH`                 | Path for Prometheus metrics endpoint                                            |           | `/metrics`                  |
| `LIGHTDASH_PROMETHEUS_PREFIX`               | Prefix for metric names.                                                        |           |                             |
| `LIGHTDASH_GC_DURATION_BUCKETS`             | Buckets for duration histogram in seconds.                                      |           | `0.001, 0.01, 0.1, 1, 2, 5` |
| `LIGHTDASH_EVENT_LOOP_MONITORING_PRECISION` | Precision for event loop monitoring in milliseconds. Must be greater than zero. |           | `10`                        |
| `LIGHTDASH_PROMETHEUS_LABELS`               | Labels to add to all metrics. Must be valid JSON                                |           |                             |

Log to see prometheus endpoint is ready
Screenshot 2024-06-19 at 10 12 13

Endpoint showing metrics

prometheus metrics

Endpoint showing custom `LIGHTDASH_PROMETHEUS_PREFIX=lightdash` and `LIGHTDASH_PROMETHEUS_LABELS='{"test":"value","test2":2}'`
Screenshot 2024-06-19 at 10 30 28

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by rephus** [comment]
Should we also update render to show these metrics ? I don't think so... 
I'll test it locally

**Comment by rephus** [comment]
Is this `process_cpu_user_seconds_total` total enough to measure the CPU usage ? I was expecting to see a % real time usage so we don't have to calculate anything...

**Comment by rephus** [comment]
Are we using `LIGHTDASH_PROMETHEUS_LABEL` to separate between api and scheduler metrics in pods ?

**Review by rephus — APPROVED** [review]

File: packages/backend/src/prometheus.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        prometheus.collectDefaultMetrics({
+            ...rest,
+        });
+        const app = express();
+        const server = http.createServer(app);
+        app.get(path, async (req, res) => {
+            res.set('Content-Type', prometheus.register.contentType);
+            res.end(await prometheus.register.metrics());
+        });
+        server.listen(port);
```
Comment (owlas) [inline packages/backend/src/prometheus.ts:23]: This looks like an uncaught exception

There's also no sigterm handling!

**Pull request merged**